core/jobs/batch_jobs/topic_migration_jobs.py

```diff
@@ -0,0 +1,362 @@
+    models.Names.BASE_MODEL, models.Names.TOPIC])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MigrateTopicJob(base_jobs.JobBase):
+    """Job that migrates topic models."""
+
+    @staticmethod
+    def _migrate_topic(
+        topic_id: str,
+        topic_model: topic_models.TopicModel
+    ) -> result.Result[Tuple[str, topic_domain.Topic], Tuple[str, Exception]]:
+        """Migrates topic and transform topic model into topic object.
+
+        Args:
+            topic_id: str. The id of the topic.
+            topic_model: TopicModel. The topic model to migrate.
+
+        Returns:
+            Result((str, Topic), (str, Exception)). Result containing tuple that
+            consist of topic ID and either topic object or Exception. Topic
+            object is returned when the migration was successful and Exception
+            is returned otherwise.
+        """
+        try:
+            topic = topic_fetchers.get_topic_from_model(topic_model)
+            topic.validate()
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((topic_id, e))
+
+        return result.Ok((topic_id, topic))
+
+    @staticmethod
+    def _generate_topic_changes(
+        topic_id: str,
+        topic_model: topic_models.TopicModel
+    ) -> Iterable[Tuple[str, topic_domain.TopicChange]]:
+        """Generates topic change objects. Topic change object is generated when
+        schema version for some field is lower than the latest schema version.
+
+        Args:
+            topic_id: str. The ID of the topic.
+            topic_model: TopicModel. The topic for which to generate the change
+                objects.
+
+        Yields:
+            (str, TopicChange). Tuple containing Topic ID and topic change
+            object.
+        """
+        subtopic_version = topic_model.subtopic_schema_version
+        if subtopic_version < feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION:
+            topic_change = topic_domain.TopicChange({
+                'cmd': (
+                    topic_domain.CMD_MIGRATE_SUBTOPIC_SCHEMA_TO_LATEST_VERSION),
+                'from_version': subtopic_version,
+                'to_version': feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION
+            })
+            yield (topic_id, topic_change)
+
+    @staticmethod
+    def _delete_topic_from_cache(
+        topic: topic_domain.Topic
+    ) -> result.Result[str, Exception]:
+        """Deletes topic from cache.
+
+        Args:
+            topic: Topic. The topic which should be deleted from cache.
+
+        Returns:
+            Result(str, Exception). The id of the story when the deletion was
+            successful or Exception when the deletion failed.
+        """
+        try:
+            caching_services.delete_multi(
+                caching_services.CACHE_NAMESPACE_TOPIC, None, [topic.id])
+            return result.Ok(topic.id)
+        except Exception as e:
+            return result.Err(e)
```

Comment:
This should not be needed anymore as I've introduced a cache flush after every job run.

Reply:
@vojtechjelinek We won't be generating Job run result for cache deletion then?

Reply:
yep we won't

Reply:
Okay. Removed